toolchain/lower/file_context.cpp

```diff
@@ -374,14 +374,11 @@ auto FileContext::BuildDISubprogram(const SemIR::Function& function,
       function.definition_id,
       [](DiagnosticLoc, const Internal::DiagnosticBase<>&) {});
   llvm::StringRef name;
-  if (function.name_id.is_valid()) {
-    name = *sem_ir().names().GetAsStringIfIdentifier(function.name_id);
+  if (auto maybe_name =
+          sem_ir().names().GetAsStringIfIdentifier(function.name_id)) {
+    name = *maybe_name;
   }
```

Comment:
I'd thought you were going to make this CHECK it had a name, i.e., something like:

```suggestion
  auto name = *sem_ir().names().GetAsStringIfIdentifier(function.name_id);
  CARBON_CHECK(name) << "Unexpected special name for function: "
                     << function.name_id;
```

Reply:
Ah, right - yep, done.